docs/docs/guides/integrate/identity-providers/azure-ad.md

```diff
@@ -0,0 +1,112 @@
+
+## Azure AD Configuration
+
+You need to have access to an AzureAD Tenant. If you do not yet have one follow [this guide from Microsoft](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant) to create one for free.
+
+### Register a new client
+
+1. Browse to the [App registration menus create dialog](https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/CreateApplicationBlade/quickStartType~/null/isMSAApp~/false) to create a new app.
+2. Give the application a name and choose who should be able to login (Single-Tenant, Multi-Tenant, Personal Accounts, etc.) This setting will also have an impact on how to configure the provider later on in ZITADEL.
+3. Choose "Web" in the redirect uri field and add the URL:
+ - {your-domain}/ui/login/login/externalidp/callback
+ - Example redirect url for the domain `https://acme-gzoe4x.zitadel.cloud` would look like this: `https://acme-gzoe4x.zitadel.cloud/ui/login/login/externalidp/callback`
+5. Save the Application (client) ID and the Directory (tenant) ID from the detail page
+
+![Azure App Registration](/img/guides/azure_app_registration.png)
+
+![Azure Client ID and Tenant ID](/img/guides/azure_client_tenant_id.png)
+
+### Add client secret
+
+To be able to authenticate your Azure application you have to generate a new client secret.
+
+1. Click on client credentials on the detail page of the application or use the menu "Certificates & secrets"
+2. Click on "+ New client secret" and enter a description and an expiry date, add the secret afterwards
+3. Copy the value of the secret. You will not be able to see the value again afterwards 
+
+![Azure Client Secret](/img/guides/azure_client_secret.png)
+
+### Token configuration
+
+To allow ZITADEL to get the information from the authenticating user you have to configure what kind of optional claims should be returned in the token.
+
+1. Click on Token configuration in the side menu
+2. Click on "+ Add optional claim"
+3. Add email, family_name, given_name and preferred_username to the id token
+
+![Azure Token configuration](/img/guides/azure_token_configuration.png)
+
+### API persmission
+
+To be able to get all the information that ZITADEL needs, you have to configure the correct permissions.
+
+1. Go to "API permissions" in the side menu
+2. Make sure the permissions include "Microsoft Graph": email, profile and User.Read
+3. The "Other permissions granted" should include "Microsoft Graph: openid"
+
+![Azure API permissions](/img/guides/azure_api_permissions.png)
+
+## ZITADEL Configuration
+
+### Create new Azure AD Provider
+
+Go to the settings of your ZITADEL instance or the organization where you like to add a new **Azure AD** provider.
+Choose the **Microsoft** provider template. This template has everything you need preconfigured. You only have to add the client ID and secret, you have created in the step before.
```

Comment:
Let's add (unrendered) line breaks after sentences for better raw markdown readability

```suggestion
Choose the **Microsoft** provider template.
This template has everything you need preconfigured.
You only have to add the client ID and secret, you have created in the step before.
```